### PR TITLE
Add Jenkins CI

### PR DESCRIPTION
**Pull request by LucioFranco**
## Description

Move to jenkins ci!

## PR Checklist

By placing an x in the boxes I certify that I have:

- [x] Acknowledged that by making this pull request I release this code under an MIT/Apache 2.0 dual licensing scheme.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+			    label 'docker'
+			} 
+                    }
+                    steps {
+                        echo 'Beginning tests...'
+                        sh 'cargo test --all'
+                        echo 'Tests done!'
+                    }
+                }
+                // stage("Test on macOS") {
```
Comment (fhaynes) [inline Jenkinsfile]:
```suggestion
                // macOS is commented out due to needing to upgrade the OS, but MacStadium did not do the original install with APFS so we cannot upgrade easily
                // stage("Test on macOS") {
```

**Comment by LucioFranco** [comment]
bors p=1000 r=fhaynes

**Comment by LucioFranco** [comment]
bors r=fhaynes

**Pull request merged**